tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala

```diff
@@ -4066,4 +4066,39 @@ class ControlSyntaxSuite extends BaseDottySuite {
     runTestAssert[Stat](code, layout)(tree)
   }
 
+  test("metals#6552: huge nested `if`") {
+    val nestedNum = 100 // was previously failing around 25-30 deep
```

Comment:
i tried 1000 and got stack overflow. not sure if should worry about that case.

Reply:
Stack overflow is fine, people can always increase it if needed.